In [2]:
import pandas as pd

# just like import and then find the ones that are special, classify what part is special
# then print
movement = pd.read_csv("movement_checked.csv")
age = pd.read_csv("age_checked.csv")

movement_ids = set(movement["ID"])
movement_flagged = set(movement.loc[movement["movement_flag"], "ID"])

age_usable = set(age.loc[age["result"] != "not enough to compare", "ID"])

age_flagged = set(age.loc[age["result"].isin(["flagged younger", "flagged older"]), "ID"])

both = movement_flagged & age_flagged

print("The IDs flagged in both were: ", sorted(both))

age_only = []
age_no_movement = []

for cluster_id in age_flagged:
    if cluster_id not in movement_ids:
        age_no_movement.append(cluster_id)
    elif cluster_id not in movement_flagged:
        age_only.append(cluster_id)

movement_only = []
movement_no_usable_age = []

for cluster_id in movement_flagged:
    if cluster_id not in age_usable:
        movement_no_usable_age.append(cluster_id)
    elif cluster_id not in age_flagged:
        movement_only.append(cluster_id)

print("Age was flagged while movement measured but not flagged:", sorted(age_only))
print("Age was flagged while movement unavailable:", sorted(age_no_movement))
print("Movement was flagged while age checked but not flagged:", sorted(movement_only))
print("Movement was flagged while no usable age comparison was found:", sorted(movement_no_usable_age))

The IDs flagged in both were:  ['NGC 1851']
Age was flagged while movement measured but not flagged: ['NGC 6171', 'Pal 12']
Age was flagged while movement unavailable: []
Movement was flagged while age checked but not flagged: ['NGC 3201', 'NGC 6101', 'NGC 6205', 'NGC 6584', 'NGC 6681', 'NGC 6934', 'NGC 6981']
Movement was flagged while no usable age comparison was found: ['2MS-GC02', 'Djorg 1', 'NGC 5139', 'NGC 6287', 'NGC 6333', 'NGC 6528', 'NGC 7006']


## Comparing the age and movement results

I matched the clusters by their IDs in both datasets. NGC 1851 was the only cluster flagged in both. Its measured movement is among the highest in the Harris sample, and its age stands out in the Krause data. But van den Berg gives it a different age, so the age evidence for this cluster isn't settled.

Pal 12 and NGC 6171 were both flagged for age, but not for movement. Pal 12 has a similar young age in both age datasets, while NGC 6171's age flag changes when we adjust the rule. Not having a movement flag does not rule out either cluster. It just means their measured speeds did not meet the cutoff I explained earlier.

Seven movement-flagged clusters had a usable age comparison but were not flagged for age. Another seven had no usable age comparison, so I can't say whether their ages are unusual under my rule. All of the age-flagged clusters did have a movement measurement.

Overall, NGC 1851 is the clearest overlap to look into, but even that does not establish that it came from another galaxy. The other groups are useful too, especially where an age measurement is missing or there aren't enough similar clusters to compare.